src/it/java/com/google/checkstyle/test/chapter4formatting/rule413emptyblocks/EmptyBlockTest.java

```diff
@@ -92,4 +93,14 @@ public void testEmptyBlockCatch() throws Exception {
         verify(checkConfig, filePath, expected, warnList);
     }
 
+    @Test
+    public void testEmptyBlockRightCurlyAlone() throws Exception {
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+        final Configuration checkConfig = getModuleConfig("EmptyBlock");
```

Comment:
Why this check for this issue? It doesn't have multiple modules from the config.
https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml#L68
Issue talks about RightCurly, so I would expect to see a test for that unless you can explain what i am missing.
https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml#L85-L104

Reply:
@rnveach, I had taken reference from the previous PR, Do I have to write test cases for RightCurly as it has multiple modules?
Please guide me, this is the first time I am working on these files.

Reply:
You need to either write tests or write usages showing that this change is being used somewhere and it is working appropriately.

From what I have seen, nothing is using `getModuleByIds`, so we can't see that your changes are working and behaving appropriately.

Reply:
```
    @Test
    public void testEmptyBlockRightCurlyAlone() throws Exception {
        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
        final Configuration checkConfig = (Configuration) getModuleByIds(" ");
        final String filePath = getPath("InputRightCurlyAloneOrEmptyNoViolations.java");

        final Integer[] warnList = getLinesWithWarn(filePath);
        verify(checkConfig, filePath, expected, warnList);
    }
```
Is writing tests like this in file EmptyBlockTest by using getModuleByIds would be fine?

Reply:
> getModuleByIds(" ");

Not sure how an empty string tests this. But you should be using a module that has multiple ids for, and use an input file that will have violations from both ids. That is how you will show its working.

Reply:
As RightCurly have multiple ids, So I should use that code in the input file which has violation from both(shown below), am I saying right?
```
<module name="RightCurly">
    <property name="id" value="RightCurlyAlone"/>
    <property name="option" value="alone"/>
    <property name="tokens"
     value="CLASS_DEF, METHOD_DEF, CTOR_DEF, LITERAL_FOR, LITERAL_WHILE, STATIC_INIT,
            INSTANCE_INIT, ANNOTATION_DEF, ENUM_DEF"/>
</module>
<module name="SuppressionXpathSingleFilter">
  <!-- suppresion is required till https://github.com/checkstyle/checkstyle/issues/7541 -->
  <property name="id" value="RightCurlyAlone"/>
  <property name="query" value="//RCURLY[parent::SLIST[count(./*)=1]
                                         or parent::OBJBLOCK[count(./*)=2]]"/>
</module>
```
And please clear whether changes in file AbstractGoogleModuleTestSupport.java file is done properly? After writing the input file then I have to write tests or write usages to show changes are working appropriately, right?

Reply:
> So I should use that code in the input file which has violation from both(shown below), am I saying right?

RightCurly is different as the second ID is a suppression, which does not print violations it hides them from the other. If we were talking 2 versions of the same check, then what you said would be right.

Also look at main issue's comment:
> we need to load together for each test

So all current RightCurlyCheck's must load both IDs for it's tests. There should be proof that both are loaded, as the suppression will start to hide existing violations being printed.

> please clear whether changes in file AbstractGoogleModuleTestSupport.java file is done properly?

I cannot say because you have no usages of your changes in the test support. Having a usage will tell you if things are working right or not.

Since I mentioned existing tests, if we are not sure which one will have violations suppressed, then create your own input that will prove you are loading both ids and a violation and a suppression are done. If the suppression didn't work, you would see an extra violation. If the violation didn't work, you would see it missing.

I would start with creating your own input. Show me you created an example input that will satisfy the results with 2 items in the config mentioned above. Use https://checkstyle.org/report_issue.html#How_to_report_a_bug.3F as a template and post it to the first post in the PR.

The code does look to be in the right direction outside of my other comments, but without a usage we can't say for sure if they are right or not.

> After writing the input file then I have to write tests or write usages to show changes are working appropriately, right?

Yes.